src/components/MAPI/clusters/CreateCluster/schemaUtils.ts

```diff
@@ -0,0 +1,173 @@
+import { GenericObjectType, UiSchema } from '@rjsf/utils';
+import { generateUID } from 'MAPI/utils';
+import { compare } from 'utils/semver';
+
+import ClusterNameWidget from './ClusterNameWidget';
+
+const uiSchemaProviderAWS: Record<string, GenericObjectType> = {
+  v0: {
+    'ui:options': {
+      order: [
+        'clusterName',
+        'clusterDescription',
+        'organization',
+        'controlPlane',
+        '*',
+      ],
+    },
+    clusterName: {
+      'ui:options': {
+        widget: ClusterNameWidget,
+      },
+    },
+  },
+};
+
+const uiSchemaProviderAzure: Record<string, GenericObjectType> = {
+  v0: {
+    'ui:options': {
+      order: [
+        'clusterName',
+        'clusterDescription',
+        'organization',
+        'controlPlane',
+        '*',
+      ],
+    },
+    clusterName: {
+      'ui:options': {
+        widget: ClusterNameWidget,
+      },
+    },
+  },
+};
+
+const uiSchemaProviderCloudDirector: Record<string, GenericObjectType> = {
+  v0: {
+    'ui:options': {
+      order: [
+        'clusterDescription',
+        'cluster',
+        'organization',
+        'controlPlane',
+        '*',
+      ],
+    },
+    clusterName: {
+      'ui:options': {
+        widget: ClusterNameWidget,
+      },
+    },
+  },
+};
+
+const uiSchemaProviderGCP: Record<string, GenericObjectType> = {
+  v0: {
+    'ui:options': {
+      order: [
+        'clusterName',
+        'clusterDescription',
+        'organization',
+        'controlPlane',
+        '*',
+      ],
+    },
+    clusterName: {
+      'ui:options': {
+        widget: ClusterNameWidget,
+      },
+    },
+  },
+};
+
+const uiSchemaProviderOpenStack: Record<string, GenericObjectType> = {
+  v0: {
+    'ui:options': {
+      order: [
+        'clusterName',
+        'clusterDescription',
+        'organization',
+        'controlPlane',
+        '*',
+      ],
+    },
+    clusterName: {
+      'ui:options': {
+        widget: ClusterNameWidget,
+      },
+    },
+  },
+};
+
+const uiSchemaProviderVSphere: Record<string, GenericObjectType> = {
+  v0: {
+    'ui:options': {
+      order: ['cluster', 'controlPlane', '*'],
+    },
+    cluster: {
+      'ui:options': {
+        order: ['name', 'organization', '*'],
+      },
+      name: {
+        'ui:options': {
+          widget: ClusterNameWidget,
+        },
+      },
+    },
+  },
+};
+
+export const prototypeProviders = [
+  'AWS',
+  'Azure',
+  'Cloud Director',
+  'GCP',
+  'Open Stack',
+  'VSphere',
+] as const;
+
+export type PrototypeProviders = typeof prototypeProviders[number];
+
+export const getDefaultFormData = (
+  provider: PrototypeProviders,
+  organization: string
+) => {
+  switch (provider) {
+    case 'VSphere':
+      return {
+        cluster: {
+          name: generateUID(5),
+        },
+      };
+    default:
+      return {
+        clusterName: generateUID(5),
+        organization,
+      };
```

Comment:
What about CloudDirector schema? There is no `clusterName` field, or am I missing something?

Reply:
Good catch - updating the cases here 👍